blog/prometheus-connector.md

```diff
@@ -0,0 +1,210 @@
+You should see the topology of your network, perhaps something like the following image.
+
+<p class="center">
+  <a href="/assets/images/blog/capture-future-1.png" data-lightbox="Skydive Topology" data-title="Skydive topology view">
+    <img src="/assets/images/blog/capture-future-1.png"/>
+  </a>
+</p>
+
+
+Make sure you are on the `Captures` view and press on `Create`.
+
+Fill in the `Targets` fields by placing the cursor in the first `Interface` field and then use the mouse to point to your network endpoint that you want to capture.
+To capture a particular flow end-to-end, fill in the second `Interface` field in a similar manner with the other network endpoint.
+Then press the `Start` button on the GUI.
+
+Alternatively, you can start the capture of flows from the command line with a command like the following:
+
+```
+skydive client capture create --gremlin "G.V().Has('Type', 'device', 'Name', 'eth0')" --type pcap 
+```
+
+The command specifies to capture all flows that match the gremlin expression - in this case, all devices that have name 'eth0'.
+
+### Generate Flows
+
+Generate some network traffic on the interfaces you specified to capture.
+For example, run some iperf traffic between entities and capture the flow via the Skydive GUI.
+
+To install `iperf3` (on ubuntu/debian):
+
+```
+sudo apt install iperf3
+```
+
+Run iperf server:
+
+```
+iperf3 -s
+```
+
+Either on the same machine or on another machine, run the iperf client:
+
+```
+iperf3 -c <address-of-iperfserever> -t 10000
+```
+
+It is also possible to generate network traffic using the inject feature of Skydive, either from the command line or through the Skydive GUI.
+Go the `Generator` page and specify the source and destination nodes between which to generate the network traffic, as well as the characterization of the flow you want to generate.
+
+
+## Prometheus setup
+
+In the Prometheus config file, add a stanza specifying the location and port used by the Skydive-Prometheus connector.
+
+```
+  - job_name: 'skydive-connector'
+    static_configs:
+    - targets: ['xx.xx.xx.xx:9100']
+
+```
+Restart Prometheus to pick up the new stanza in the config file.
+
+## Verify Operation
+
+Wait for 60 seconds and then check that we have data arriving in the Prometheus GUI.
+Look for a metric beginning with `skydive` such as `skydive_network_connection_total_bytes`.
+You can display a graph of all the flows that have been captured that report the `skydive_network_connection_total_bytes` metrics.
+To see a graph of the bandwidth (bytes per second instead of total bytes transferred) used by flows, you can graph on the variable `rate(skydive_network_connection_total_bytes[1m])`.
+
+The data shown in the graph should contain the following fields:
+  * `initiator_ip`
+  * `target_ip`
+  * `initiator_port`
+  * `target_port`
+  * `direction`
+  * `node_tid`
+
+In addition, each such tuple is associated with timestamps and corresponding values that appear in the graph.
+
+##
```

Comment:
Is there any missing content here ? It seems to be a title

Reply:
Addressed reviewers comments. Pushed commits with updates.

Reply:
Extra heading was removed.
Other edit suggestions incorporated in update.